src/Utilities/SidreDataCollectionInline.hh

```diff
@@ -0,0 +1,531 @@
+      view_count++;
+   }
+   return m_datastore_ptr->getRoot()->getView(view_name + "0");
+}
+
+template<typename Dimension>
+inline
+axom::sidre::View *SidreDataCollection::alloc_view(const std::string &view_name, 
+                                                   const Spheral::Field<Dimension, Dim<3>::Tensor> &field)
+{
+   axom::sidre::DataTypeId dtype = field.getAxomType();
+   axom::IndexType num_elements = 9;
+   int view_count = 0;
+   for (u_int i = 0; i < field.size(); i++)
+   {
+      auto *data = &(*field[i].begin());
+      m_datastore_ptr->getRoot()->createView(view_name + std::to_string(view_count), dtype, 
+                                                         num_elements, (void*)data);
+      view_count++;
+   }
+   return m_datastore_ptr->getRoot()->getView(view_name + "0");
+}
+
+template<typename Dimension>
+inline
+axom::sidre::View *SidreDataCollection::alloc_view(const std::string &view_name, 
+                                                   const Spheral::Field<Dimension, Dim<3>::SymTensor> &field)
+{
+   axom::sidre::DataTypeId dtype = field.getAxomType();
+   axom::IndexType num_elements = 3;
+   int view_count = 0;
+   for (u_int i = 0; i < field.size(); i++)
+   {
+      auto *data = &(*field[i].begin());
+      m_datastore_ptr->getRoot()->createView(view_name + std::to_string(view_count), dtype, 
+                                                         num_elements, (void*)data);
+      view_count++;
+   }
+   return m_datastore_ptr->getRoot()->getView(view_name + "0");
+}
+
+template<typename Dimension>
+inline
+axom::sidre::View *SidreDataCollection::alloc_view(const std::string &view_name, 
+                                                   const Spheral::Field<Dimension, Dim<3>::ThirdRankTensor> &field)
+{
+   axom::sidre::DataTypeId dtype = field.getAxomType();
+   axom::IndexType num_elements = 1;
+   int view_count = 0;
+   for (u_int i = 0; i < field.size(); i++)
+   {
+      auto *data = &(*field[i].begin());
+      m_datastore_ptr->getRoot()->createView(view_name + std::to_string(view_count), dtype, 
+                                                         num_elements, (void*)data);
+      view_count++;
+   }
+   return m_datastore_ptr->getRoot()->getView(view_name + "0");
+}
+
+template<typename Dimension>
+inline
+axom::sidre::View *SidreDataCollection::alloc_view(const std::string &view_name, 
+                                                   const Spheral::Field<Dimension, Dim<3>::FourthRankTensor> &field)
+{
+   axom::sidre::DataTypeId dtype = field.getAxomType();
+   axom::IndexType num_elements = 1;
+   int view_count = 0;
+   for (u_int i = 0; i < field.size(); i++)
+   {
+      auto *data = &(*field[i].begin());
+      m_datastore_ptr->getRoot()->createView(view_name + std::to_string(view_count), dtype, 
+                                                         num_elements, (void*)data);
+      view_count++;
+   }
+   return m_datastore_ptr->getRoot()->getView(view_name + "0");
+}
```

Comment:
The majority of the alloc_view methods are pretty much identical and can be templated together, the only difference I see in most of them is the value of num_elements which can be extracted from the Fields DataType type trait. Templating this will greatly reduce code duplication here